ingestion_server/ingestion_server/cleanup.py

```diff
@@ -133,12 +136,14 @@ def cleanup_tags(tags):
         if not tags:
             return None
         for tag in tags:
-            below_threshold = False
+            alt_filtered = False
```

Comment:
This code has been bothering me for a long time because it's difficult to understand all branches. What if we convert it to something like this:

```py
if not isinstance(tag.get("name"), str) or "accuracy" in tag and float(tag["accuracy"]) < TAG_MIN_CONFIDENCE or
tag.get("provider") in TAG_SKIP_PROVIDERS:
    should_filter = True
else:
    lower_tag = tag["name"].lower()
    should_filter = _tag_denylisted(lower_tag)

```
This will also be a bit better for performance because if the tag is `{}`, we won't check for it's provider and accuracy.

Reply:
This is a good question - hopefully this code is going away long-term, but I'll try and see if I can simplify/optimize it without spending too much time on it :)

Reply:
Looking at this again, we log specifically if the tag is _malformed_ based on the current set of conditions, but then don't log anything if it just gets filtered. So I think, given this will eventually be revised, I'll just leave it as-is for now.

Reply:
Do we really need to log that? Malformed in this case means that a tag is either not a dictionary or doesn't have a string name value, right?
I think we've already removed those from the catalog db, haven't we?

Reply:
You're right that I think we have - that said, I'm already working on #4541 and have altered the logic in these steps anyway for that change. If you have a blocking objection here than I can change it, otherwise I'd rather wait until we have to rework these cleanup steps entirely as part of the data refresh removal!